lib/routes/account.js

```diff
@@ -840,6 +840,17 @@ module.exports = function (
         log.begin('Account.device', request)
         var payload = request.payload
         var sessionToken = request.auth.credentials
+        // Clients have been known to send spurious device updates,
+        // which generates lots of unnecessary database load.
+        // Don't write out the update if nothing has actually changed.
+        if (payload.id && payload.id === sessionToken.deviceId.toString('hex') &&
```

Comment:
@rfk in https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#post-v1accountdevice 

`If no device id is specified, both name and type must be provided. ` - will this fail when `id` is missing?